src/sentry/static/sentry/app/views/performance/trends/chart.tsx

```diff
@@ -0,0 +1,212 @@
+import {Series} from 'app/types/echarts';
+import theme from 'app/utils/theme';
+
+import {trendToColor, getIntervalRatio} from './utils';
+import {TrendChangeType, TrendsStats} from './types';
+import {HeaderTitleLegend} from '../styles';
+
+const QUERY_KEYS = [
+  'environment',
+  'project',
+  'query',
+  'start',
+  'end',
+  'statsPeriod',
+] as const;
+
+type ViewProps = Pick<EventView, typeof QUERY_KEYS[number]>;
+
+type Props = WithRouterProps &
+  ViewProps & {
+    api: Client;
+    location: Location;
+    organization: OrganizationSummary;
+    trendChangeType: TrendChangeType;
+    transaction?: string;
+    isLoading: boolean;
+    statsData: TrendsStats;
+  };
+
+function getChartTitle(trendChangeType: TrendChangeType): string {
+  switch (trendChangeType) {
+    case TrendChangeType.IMPROVED:
+      return t('Most Improved');
+    case TrendChangeType.REGRESSION:
+      return t('Worst Regressions');
+    default:
+      throw new Error('No trend type passed');
+  }
+}
+
+function transformEventStats(data: EventsStatsData, seriesName?: string): Series[] {
+  return [
+    {
+      seriesName: seriesName || 'Current',
+      data: data.map(([timestamp, countsForTimestamp]) => ({
+        name: timestamp * 1000,
+        value: countsForTimestamp.reduce((acc, {count}) => acc + count, 0),
+      })),
+    },
+  ];
+}
+
+function getIntervalLine(series: Series[], intervalRatio: number) {
+  if (!series.length || !series[0].data || !series[0].data.length) {
+    return [];
+  }
+  const additionalLineSeries = [
+    {
+      data: [],
+      markLine: {
+        data: [] as any[],
+        label: {show: false},
+        lineStyle: {
+          normal: {
+            color: theme.gray700,
+            type: 'dashed',
+            width: 2,
+          },
+        },
+        symbol: ['none', 'none'],
+        tooltip: {
+          show: false,
+        },
+      },
+      seriesName: 'Split',
+    },
+  ];
+  const seriesStart = parseInt(series[0].data[0].name as string, 0);
+  const seriesEnd = parseInt(series[0].data.slice(-1)[0].name as string, 0);
+
+  if (additionalLineSeries && seriesEnd > seriesStart) {
+    const seriesDiff = seriesEnd - seriesStart;
+    const seriesLine = seriesDiff * (intervalRatio || 0.5) + seriesStart;
+    additionalLineSeries[0].markLine.data = [
+      {
+        value: 'Comparison line',
+        xAxis: seriesLine,
+      },
+    ];
+  }
+  return additionalLineSeries;
+}
+
+class Chart extends React.Component<Props> {
+  render() {
+    const props = this.props;
+
+    const {
+      trendChangeType,
+      router,
+      statsPeriod,
+      project,
+      environment,
+      transaction,
+      statsData,
+      isLoading,
+    } = props;
+    const chartTitle = getChartTitle(trendChangeType);
+    const lineColor = trendToColor[trendChangeType];
+
+    const events = statsData && statsData[transaction || ''];
+    const data = events ? events.data : [];
```

Comment:
No big deal but you can do this in typescript with `events?.data ?? []`

Reply:
ty ty 👍